### PR TITLE
Update AppServiceComboBox to adapt AzureComboBox refresh strategy

### PR DESCRIPTION
**Pull request by Flanker32**
- Update AppServiceComboBox to adapt AzureComboBox refresh strategy
- Add model from configuration during `loadItems` for draft item

> Known Issues: Can not select default value while combo box is refreshing

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java [hunk 1]
```diff
@@ -12,72 +12,45 @@
...
+    @NotNull
     @Override
-    protected void handleLoadingError(final Throwable e) {
-        final Throwable rootCause = ExceptionUtils.getRootCause(e);
-        if (rootCause instanceof InterruptedIOException || rootCause instanceof InterruptedException) {
-            // Swallow interrupted exception caused by unsubscribe
-            return;
+    protected List<? extends T> loadItems() throws Exception {
+        final List<T> items = loadAppServiceModels();
+        if (super.getItemReference() != null && configModel != null && configModel.isNewCreateResource()) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:43]: remove super.

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:43]: this is a general problem, we shall not resolve the combo box problem in AppServiceComboBox but AzureComboBox

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java [hunk 2]
```diff
@@ -121,6 +121,10 @@ public void setValue(final ItemReference<T> val) {
         this.refreshValue();
     }
 
+    protected ItemReference<T> getItemReference() {
+        return this.value instanceof ItemReference ? (ItemReference<T>) this.value : null;
+    }
+
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java:127]: remove this method. and use `this.value` directly in subclasses.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java [hunk 3]
```diff
@@ -12,72 +12,45 @@
...
+            final T target = items.stream().filter(model -> getItemReference().is(model)).findAny().orElse(null);
+            if (target == null) {
+                items.add(configModel);
+            }
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java:47]: if I didn't misunderstand, this block can be simplified to :
```
final boolean exists = items.stream().anyMatch(i->AppServiceComboBoxModel.isSameApp(i, configModel))
if(exists){
    items.add(configModle);
}
```

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java (hunk 1 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java]: `getItemReference` removed, only process it in `AppServiceComboBox`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java (hunk 2 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureComboBox.java]: Yes, will set the `ItemReference` in `AppServiceComboBox`, and remove the `getItemReference` in `AzureComboBox`

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**